src/openrct2/scripting/ScNetwork.hpp

```diff
@@ -0,0 +1,377 @@
+        }
+    };
+
+    class ScPlayer
+    {
+    private:
+        int32_t _id;
+
+    public:
+        ScPlayer(int32_t id)
+            : _id(id)
+        {
+        }
+
+        int32_t id_get()
+        {
+            return _id;
+        }
+
+        std::string name_get()
+        {
+#    ifndef DISABLE_NETWORK
+            auto index = network_get_player_index(_id);
+            if (index == -1)
+                return {};
+            return network_get_player_name(index);
+#    else
+            return {};
+#    endif
+        }
+
+        int32_t group_get()
+        {
+#    ifndef DISABLE_NETWORK
+            auto index = network_get_player_index(_id);
+            if (index == -1)
+                return {};
+            return network_get_player_group(index);
+#    else
+            return 0;
+#    endif
+        }
+        void group_set(int32_t value)
+        {
+#    ifndef DISABLE_NETWORK
+            auto playerSetGroupAction = PlayerSetGroupAction(_id, value);
+            GameActions::Execute(&playerSetGroupAction);
+#    endif
+        }
+
+        int32_t ping_get()
+        {
+#    ifndef DISABLE_NETWORK
+            auto index = network_get_player_index(_id);
+            if (index == -1)
+                return {};
+            return network_get_player_ping(index);
+#    else
+            return 0;
+#    endif
+        }
+
+        int32_t commandsRan_get()
+        {
+#    ifndef DISABLE_NETWORK
+            auto index = network_get_player_index(_id);
+            if (index == -1)
+                return {};
+            return network_get_player_commands_ran(index);
+#    else
+            return 0;
+#    endif
+        }
+
+        int32_t moneySpent_get()
+        {
+#    ifndef DISABLE_NETWORK
+            auto index = network_get_player_index(_id);
+            if (index == -1)
+                return {};
+            return network_get_player_money_spent(index);
+#    else
+            return 0;
+#    endif
+        }
+
+        static void Register(duk_context* ctx)
+        {
+            dukglue_register_property(ctx, &ScPlayer::id_get, nullptr, "id");
+            dukglue_register_property(ctx, &ScPlayer::name_get, nullptr, "name");
+            dukglue_register_property(ctx, &ScPlayer::group_get, &ScPlayer::group_set, "group");
+            dukglue_register_property(ctx, &ScPlayer::ping_get, nullptr, "ping");
+            dukglue_register_property(ctx, &ScPlayer::commandsRan_get, nullptr, "commandsRan");
+            dukglue_register_property(ctx, &ScPlayer::moneySpent_get, nullptr, "moneySpent");
+        }
+    };
+
+    class ScNetwork
+    {
+    private:
+#    ifdef __clang__
+        [[maybe_unused]]
+#    endif
+        duk_context* _context;
+
+    public:
+        ScNetwork(duk_context* ctx)
+            : _context(ctx)
+        {
+        }
+
+        std::string mode_get()
+        {
+#    ifndef DISABLE_NETWORK
+            switch (network_get_mode())
+            {
+                default:
```

Comment:
I usually avoid adding the `default:` for switches on enumerations. That's because when you add a new enumeration item, the compiler will fail to compile because one enumeration item is missing. Then you know every place you gotta change.

```suggestion
```